### ISSUE TITLE
[BUG][Fuzzer][wrong-code] `T.copy` of a column-padded shared tile silently packs the wrong pitch (interleaved rows) not the declared stride

### ISSUE THREAD

**Issue by Hughshine** [tile-ai/tilelang#2581]
### Required prerequisites

- [x] I have read the documentation <https://tilelang.com>.
- [x] I have searched the [Issue Tracker](https://github.com/tile-ai/tilelang/issues) that this hasn't already been reported. (comment there if it has.)

### What version of TileLang are you using?

0.1.11

### System information

- TileLang `0.1.11`; source pinned to tag `v0.1.11` (commit `cd37ed5fc35ae7a60a1277c8eb49028174ac51e6`), byte-identical to the installed wheel.
- Python 3.12.3, linux; PyTorch 2.12.1+cu130, CUDA 13.0.
- GPU: **NVIDIA H100 80GB HBM3** (compute capability 9.0, sm_90). **Reproduced at runtime on this H100.** Hopper-specific: the `cp.async.bulk` (1-D bulk TMA) lowering of these copies is only selected on sm_90.

### Problem description

A `T.copy` whose shared operand is a **column-padded sub-region** — `As = T.alloc_shared((BM, SBN), ...)` with `SBN > BN` (innermost / column padding, a common idiom to satisfy GEMM swizzle or avoid shared-bank conflicts), copying only the `As[:, :BN]` sub-tile — compiles and runs with no error but **silently returns wrong numbers**. The copy is admitted to the 1-D bulk-TMA path and lowered to a `cp.async.bulk` that packs `BM*BN` elements **contiguously** (effective pitch `BN`) from `&As[0]`, while every ordinary `As[i, j]` access uses the **declared** pitch `SBN`. Rows therefore land at the wrong offsets: the output is the correct rows interleaved with the padding gap (row 0 correct, row 1 = padding, row 2 = source row 1, …), **not** a dropped tail.

This is a **pitch mismatch, not a size problem**: it fires even when the total transfer size is a whole multiple of 16 bytes (e.g. `As(64,128)` copy `[:, :64]` fp16 = 8192 bytes). Row-padding (`SBM > BM`, more rows) is fine — rows stay contiguous; only column / innermost padding breaks.

Both directions are affected:

- **Store** (`T.copy(As[:, :BN], Out)`, shared→global) fires on the **default auto path** — no `prefer_instruction` needed — so it can surface in ordinary "stage a tile in a padded shared buffer, then store it back" code.
- **Load** (`T.copy(A, As[:, :BN])`, global→shared) fires when the 1-D bulk load is selected (`prefer_instruction="tma"`).

The same copy with the TMA path disabled (`disable_tma=True`) is bit-exact, and the same copy with no padding (`SBN == BN`) on the same TMA path is bit-exact — so the program is well-formed and supported; only the padded-sub-region bulk lowering is wrong.

### Reproducible example code

Run on an sm_90 (Hopper) GPU so the 1-D bulk-TMA path is selected. Exact-integer fp16 inputs, so any nonzero error is a real miscompile, not fp rounding.

```python
import tilelang
tilelang.disable_cache()
import tilelang.language as T
import torch

BM, BN, SBN = 64, 64, 128   # column-padded shared: alloc (64,128), copy the (64,64) sub-region

@T.prim_func
def kernel(A: T.Tensor((BM, BN), "float16"), Out: T.Tensor((BM, BN), "float16")):
    with T.Kernel(1, threads=128):
        As = T.alloc_shared((BM, SBN), "float16")   # SBN=128 > BN=64 -> innermost/column pad
        for i, j in T.Parallel(BM, BN):
            As[i, j] = A[i, j]                        # SIMT fill: correct, declared pitch SBN
        T.copy(As[:, :BN], Out)                       # shared->global: DEFAULT path -> 1-D bulk TMA store

k = tilelang.compile(kernel)
A = torch.arange(BM * BN, dtype=torch.float16, device="cuda").reshape(BM, BN)
Out = torch.zeros((BM, BN), dtype=torch.float16, device="cuda")
k(A, Out); torch.cuda.synchronize()

print("bytes transferred:", BM * BN * 2, "(% 16 =", (BM * BN * 2) % 16, ")")  # 8192, a multiple of 16
print("tma_store in CUDA:", "tma_store" in k.get_kernel_source())             # True
print("max_err:", (Out.float() - A.float()).abs().max().item())               # 4096.0  (SILENT-WRONG)
print("row 0 ok:", torch.equal(Out[0].cpu(), A[0].cpu()),                     # True
      "| row 1 ok:", torch.equal(Out[1].cpu(), A[1].cpu()))                    # False (got the padding gap)
```

Observed on H100: `bytes transferred: 8192 (% 16 = 0)`, `tma_store in CUDA: True`, `max_err: 4096.0`, `row 0 ok: True | row 1 ok: False`.

<details>
<summary>Discriminating controls — the only change between WRONG and OK is the padded-sub-region TMA lowering (H100, TESTED)</summary>

Same kernel shape, exact-integer fp16 inputs; `max_err` is 0.0 iff correct:

| edge | config | path | `max_err` |
|---|---|---|---|
| store | `As(64,128)` copy `[:, :64]` | **default** (1-D bulk `tma_store`) | **4096.0 (WRONG)** |
| store | `As(64,128)` copy `[:, :64]` | `disable_tma=True` (SIMT) | 0.0 (OK) |
| store | `As(64,64)` no pad | default (1-D bulk `tma_store`) | 0.0 (OK) |
| store | `As(8,16)` copy `[:, :8]` | default | **63.0 (WRONG)** |
| load | `As(64,128)` copy `[:, :64]` | `prefer_instruction="tma"` (1-D bulk `tma_load`) | **4096.0 (WRONG)** |
| load | `As(64,128)` copy `[:, :64]` | `disable_tma=True` (SIMT) | 0.0 (OK) |
| load | `As(64,64)` no pad | `prefer_instruction="tma"` | 0.0 (OK) |

The no-pad row uses the **identical** 1-D bulk edge and is bit-exact; the `disable_tma` row is bit-exact — isolating the defect to "1-D bulk copy of a column-padded shared sub-region," in both directions. (The 8192-byte case shows this is not a size/alignment issue: it is a multiple of 16.)

</details>

### Traceback

```pytb
No traceback — the kernel compiles and runs to completion; the result is silently wrong (rows interleaved with the padding gap).
```

### Expected behavior

A `T.copy` of a column-padded shared sub-region should either (a) emit a strided bulk transfer / per-row descriptor that respects the declared shared pitch `SBN`, so `As[i, j]` reads/writes what every other access uses, or (b) if the 1-D bulk path cannot express the sub-range pitch, fall back to the SIMT / `cp.async` copy — which is correct here, as `disable_tma=True` is bit-exact on every shape above. Silently packing with pitch `BN` while the rest of the kernel uses pitch `SBN` is the failure; the SIMT path already ships the correct behavior, so this is an internal inconsistency to fix, not a design question.

### Additional context

**Root cause.** The eligibility gate `CheckBulkCopy1D` ([`copy_analysis.cc#L268-L310`](https://github.com/tile-ai/tilelang/blob/cd37ed5fc35ae7a60a1277c8eb49028174ac51e6/src/cuda/op/copy_analysis.cc#L268-L310)) decides a copy is a single contiguous 1-D block from just three facts: `shared_is_contiguous`, `global_is_contiguous`, and `element_match`. But `shared_is_contiguous` is tested only against `makeLinearLayout(shared_tensor->shape)` — the **full** `(BM, SBN)` shape ([`#L276`](https://github.com/tile-ai/tilelang/blob/cd37ed5fc35ae7a60a1277c8eb49028174ac51e6/src/cuda/op/copy_analysis.cc#L276)) — and `shared_elements` is the product of the **sub-range** extents `shared_range[i]->extent` (= `BM*BN`, [`#L299-L302`](https://github.com/tile-ai/tilelang/blob/cd37ed5fc35ae7a60a1277c8eb49028174ac51e6/src/cuda/op/copy_analysis.cc#L299-L302)). It never verifies that the copied sub-range `As[0:BM, 0:BN]` is contiguous **inside** the `(BM, SBN)` buffer. With `SBN > BN` the sub-range is non-contiguous, yet the gate returns true, so `CheckBulkStore1D`/`CheckBulkLoad1D` ([`#L312-L330`](https://github.com/tile-ai/tilelang/blob/cd37ed5fc35ae7a60a1277c8eb49028174ac51e6/src/cuda/op/copy_analysis.cc#L312-L330)) accept it and `SelectTmaInst` ([`#L505-L522`](https://github.com/tile-ai/tilelang/blob/cd37ed5fc35ae7a60a1277c8eb49028174ac51e6/src/cuda/op/copy_analysis.cc#L505-L522)) picks the 1-D bulk path.

`LowerBulk1D` ([`copy.cc#L1975-L2049`](https://github.com/tile-ai/tilelang/blob/cd37ed5fc35ae7a60a1277c8eb49028174ac51e6/src/cuda/op/copy.cc#L1975-L2049)) then computes the base `shared_offset` from the **declared** strides ([`#L1980-L1986`](https://github.com/tile-ai/tilelang/blob/cd37ed5fc35ae7a60a1277c8eb49028174ac51e6/src/cuda/op/copy.cc#L1980-L1986)) — so `&As[0]` is correct — but issues the transfer as `total_bytes = TMATransactionBytesFromElements(elements, dtype)` of **contiguous** memory, with `elements = BM*BN` ([`#L2014`, `#L2038-L2039`](https://github.com/tile-ai/tilelang/blob/cd37ed5fc35ae7a60a1277c8eb49028174ac51e6/src/cuda/op/copy.cc#L2014-L2039)). The emitted device call is the size-only `tl::tma_store(&Out[0], &As[0], 8192, ...)` (store) / `tl::tma_load(&As[0], &A[0], mbar, 8192, ...)` (load) — no per-row stride is carried, so the padding pitch is lost and rows shift by `SBN - BN` each. Dispatch: [`copy.cc#L690-L694`](https://github.com/tile-ai/tilelang/blob/cd37ed5fc35ae7a60a1277c8eb49028174ac51e6/src/cuda/op/copy.cc#L690-L694).

**Suggested fix.** In `CheckBulkCopy1D`, keep the copy off the 1-D bulk path when the shared sub-range is not contiguous inside the declared buffer — i.e. require, for every dimension outer to the innermost, that either the range spans the full declared extent or the padded strides make the covered region truly contiguous (equivalently, that `product(shared_range extents) * dtype_bytes` equals the byte span implied by the declared strides). When that fails, fall back to the SIMT copy (as `disable_tma=True` already does correctly). This is the same class of gap that `#2525` (unaligned 1-D bulk size) hits in the same function: `CheckBulkCopy1D` verifies contiguity and element-count but not the physical byte layout of the sub-range.

**Provenance.** The `CheckBulkCopy1D` gate with exactly this structure (`shared_is_contiguous` vs `makeLinearLayout(full shape)` + `shared_elements` from sub-range extents + `element_match`, no sub-range-pitch check) was introduced in PR **#788 "[TMA] Automatically lower 1d tma in appropriate cases"** (commit [`9d7d45be`](https://github.com/tile-ai/tilelang/commit/9d7d45bebcb67bb03818cf3bb0ca041a8c5215ae), 2025-09-06), building on the 1-D TMA support from #761. The subsequent refactors (#2138, #2297) moved the code into `src/cuda/op/` but preserved the missing check.

**Not a duplicate.** Closest is `#2525` (1-D bulk store whose byte size isn't a multiple of 16 silently drops the unaligned tail) — same eligibility gate (`CheckBulkCopy1D`), but a **different missing check and a different symptom**: `#2525` drops a small tail when `total_bytes % 16 != 0`, whereas this fires when the transfer size **is** a clean multiple of 16 (8192 B) and produces **interleaved wrong rows** from a pitch shift, not a truncated tail. Distinct from the TMA-copy *crash* reports (`#2527` non-16B innermost offset → `cudaErrorIllegalInstruction`; `#2528` `block_K=32` double-buffer alignment crash; `#2529` partial-tile `IndexError` in layout inference) — those are crashes from alignment/offset/partial-tile constraints; this is a silent wrong result on a clean-compiling, clean-running kernel. Distinct from `#2543` (whole-tile `T.copy` dropping `annotate_safe_value`, a SIMT-path OOB-fill issue). Searched open + closed tile-ai/tilelang issues; no existing report covers a column-padded shared sub-region TMA copy pitch mismatch.
